app/com/linkedin/drelephant/exceptions/AnalyzedLog.java

```diff
@@ -0,0 +1,172 @@
+package com.linkedin.drelephant.exceptions;
+
+//Given a log string, analyzes and returns the relevant information!
+import org.apache.log4j.Logger;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AnalyzedLog {
+
+        private static final Logger logger = Logger.getLogger(AnalyzedLog.class);
+        private String state;
+        private ExceptionLoggingEvent exception;
+        private Set<String> list;
+        private String log;
+
+
+        public AnalyzedLog(String log){
```

Comment:
Constructor is too large. Split it! Move all the parsing code to a separate method called ```parseRawLog```.